concent_api/core/models.py

```diff
@@ -485,3 +493,478 @@ def clean(self):
+            })
+
+        if not len(self.provider_ethereum_public_key) == GOLEM_PUBLIC_KEY_LENGTH or not isinstance(self.provider_ethereum_public_key, bytes):
+            raise ValidationError({
+                'provider_ethereum_public_key': f'provider_ethereum_public_key must be bytes instance and must have length equal to {GOLEM_PUBLIC_KEY_LENGTH}'
+            })
+
+        if not self.package_hash[:4] == HASH_FUNCTION:
+            raise ValidationError({
+                'package_hash': f'package_hash should be hashed with {HASH_FUNCTION} function'
+            })
+
+        if not len(self.package_hash) == HASH_LENGTH:
+            raise ValidationError({
+                'package_hash': f'package_hash should has length equal to {HASH_LENGTH}'
+            })
+
+        if not isinstance(self.concent_enabled, bool):
+            raise ValidationError({
+                'concent_enabled': 'concent_enabled should be instance of boolean'
+            })
+
+        if not isinstance(self.price, int) or not self.price >= 0:
+            raise ValidationError({
+                'price': 'Price must be an integer and bigger than or equal 0'
+            })
+
+
+class StoredTaskFailure(AbstractMessageModel):
+
+    ITEMS = [
+        'task_to_compute',
+        'err',
+        'header',
+        'sig',
+    ]
+
+    if not len(ITEMS) == len(message.tasks.TaskFailure().__slots__):
+        raise ValidationError({
+            'StoredTaskFailure': 'StoredTaskFailure model has not same fields as golem_messages.message.tasks.TaskFailure()'
+        })
+
+    for model_item, golem_message_item in zip(ITEMS, message.tasks.TaskFailure().__slots__):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'StoredTaskFailure': 'StoredTaskFailure model has not same fields as golem_messages.message.tasks.TaskFailure()'
+            })
+
+    task_to_compute = ForeignKey(StoredTaskToCompute)
+    err = TextField()
+
+    '''
+    VALIDATION FOR err
+    '''
+
+
+class StoredCannotComputeTask(AbstractMessageModel):
+
+    ITEMS = [
+        'task_to_compute',
+        'reason',
+        'header',
+        'sig',
+    ]
+
+    if not len(ITEMS) == len(message.tasks.CannotComputeTask().__slots__):
+        raise ValidationError({
+            'StoredCannotComputeTask': 'StoredCannotComputeTask model has not same fields as golem_messages.message.tasks.CannotComputeTask()'
+        })
+
+    for model_item, golem_message_item in zip(ITEMS, message.tasks.CannotComputeTask().__slots__):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'StoredCannotComputeTask': 'StoredCannotComputeTask model has not same fields as golem_messages.message.tasks.CannotComputeTask()'
+            })
+
+    class CannotComputeTaskReason(ChoiceEnum):
+        WrongCTD = 'WrongCTD'
+        WrongKey = 'WrongKey'
+        WrongAddress = 'WrongAddress'
```

Comment:
Isn't this enum already defined in golem-messages? We should not be copying it here.

If you just want to use `.choices()` on it, you should just make a separate function (or a one-liner) that prepares the list of choices for the model instead of repeating the whole enum.

Reply:
I think the reason for that is the same as in Karol's answer:

> This validation starts when we start django server. If any message would change their structure in golem_message repo we will know it from beginning.